### ISSUE TITLE
Row headers for Tables

### ISSUE THREAD

**Issue by timgrant-philly** [ant-design/ant-design#32188]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Currently, the Ant table component doesn’t provide developers the ability to specify <th scope=”row”> on tables. This prevents us from being able to programmatically, accessibly, and usably define tables that use rowheaders.

Without row headers, users of screen readers cannot quickly identify rows, so it fails WCAG Success Criterion 1.3.1. 

Ref: https://www.w3.org/TR/WCAG20-TECHS/F91.html

Tagging given columns as row headers, that are styled appropriately, will let sighted users identify rows more quickly as well -- especially for mobile-sized screens.

Selectable rows in ANT Design do not have the ability to have labels that correspond to the controls (and thus fail WCAG 2.1 Success Criterion 2.5.3). Being able to specify a row header lays the groundwork for labeling the row selection control with the row header.

https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html

NOTE: Our corporate development team has time, budget, and approval to implement this enhancement.

### What does the proposed API look like?
Add the following attribute to ColumnType:

rowHeader?: boolean;

**Comment by zombieJ** [comment]
What DOM attribute will `rowHeader` be like? Change `td` to `th` or something else?

**Comment by timgrant-philly** [comment]
@zombieJ , we will use w3's examples as a paradigm. That suggests we will use: `<th scope="row">`.

That said, we will test the solution VoiceOver and NVDA, as well as ensuring it passes Microsoft's FastPass and WAVE Accessibility testing tools. Any adverse results from these tests would inform the implementation details.

https://www.w3.org/WAI/tutorials/tables/

**Comment by dgreene1** [comment]
@afc163 / @zombieJ I noticed that a “help wanted” tag was never added to this. Does that mean that you will or will not accept a PR that fixes this?

it’s a standard accessibility pattern with a generally accepted solution which @timgrant-philly links to above.

**Comment by mellis481** [comment]
@afc163 @zombieJ Is there a response to @dgreene1's question?

**Comment by mellis481** [comment]
FYI this was completed in rc-table [here](https://github.com/react-component/table/pull/859) and incorporated into Ant.

**Comment by afc163** [comment]
Supported in https://github.com/ant-design/ant-design/issues/39688
